### PR TITLE
travis: force python2, add --cache-dir=$PWD/cache

### PR DESCRIPTION
**Pull request by alltilla**
Our python codebase (tests) is based on python2, and it was assumed, that `python` will point to `python2`.

Newer distros (the arm based runner in travis) redirect `python` to `python3`, which was not our intention.

For some reason python3 does not have all the packages we need in pip, so our travis job breaks.

This patch forces `python2` for `pip`.

I also added `--cache-dir=$PWD/cache`, because the arm travis runner does not grant the default cache dir with proper capabilities.

Signed-off-by: Attila Szakacs <attila.szakacs@balabit.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -95,7 +95,7 @@ matrix:
       compiler: clang
       script:
         - set -e
-        - pip install --user pre-commit-hooks
+        - python2 -m pip install --user --cache-dir=$PWD/cache pre-commit-hooks
```
Comment (MrAnno) [inline .travis.yml]: `--with-python=2` is already hard-coded in travis.yml, so I think it was a good idea to use `python2` explicitly for all jobs. 

Why did you add cache-dir to the `autotools-clang` and `cmake-submodules-and-pytest` jobs?

**Comment by kira-syslogng** [comment]
Build FAILURE

File: .travis.yml (hunk 1 above)
Comment (alltilla) [reply .travis.yml]: It seemed like a non-destructive move, although it is not necessarily needed. I can remove it, if you insist.

Comment (MrAnno) [reply .travis.yml]: Thanks. I don't, I was just curious.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by lbudai** [comment]
@alltilla: thanks!
